application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java

```diff
@@ -89,70 +88,35 @@ public class ThingsboardInstallService {
-                            }
-                        case "3.6.2":
-                            log.info("Upgrading ThingsBoard from version 3.6.2 to 3.6.3 ...");
-                            databaseEntitiesUpgradeService.upgradeDatabase("3.6.2");
-                            systemDataLoaderService.updateDefaultNotificationConfigs(true);
-                        case "3.6.3":
-                            log.info("Upgrading ThingsBoard from version 3.6.3 to 3.6.4 ...");
-                            databaseEntitiesUpgradeService.upgradeDatabase("3.6.3");
-                        case "3.6.4":
-                            log.info("Upgrading ThingsBoard from version 3.6.4 to 3.7.0 ...");
-                            databaseEntitiesUpgradeService.upgradeDatabase("3.6.4");
-                            dataUpdateService.updateData("3.6.4");
-                            entityDatabaseSchemaService.createCustomerTitleUniqueConstraintIfNotExists();
-                            systemDataLoaderService.updateDefaultNotificationConfigs(false);
-                            systemDataLoaderService.updateSecuritySettings();
-                        case "3.7.0":
-                            log.info("Upgrading ThingsBoard from version 3.7.0 to 3.8.0 ...");
-                            databaseEntitiesUpgradeService.upgradeDatabase("3.7.0");
-                        case "3.8.0":
-                            log.info("Upgrading ThingsBoard from version 3.8.0 to 3.8.1 ...");
-                        case "3.8.1":
-                            log.info("Upgrading ThingsBoard from version 3.8.1 to 3.9.0 ...");
-                            databaseEntitiesUpgradeService.upgradeDatabase("3.8.1");
-                            installScripts.updateResourcesUsage();
-                            //TODO DON'T FORGET to update switch statement in the CacheCleanupService if you need to clear the cache
-                            break;
-                        default:
-                            throw new RuntimeException("Unable to upgrade ThingsBoard, unsupported fromVersion: " + upgradeFromVersion);
-                    }
+                    databaseSchemaVersionService.validateSchemaSettings();
+                    //TODO DON'T FORGET to update SUPPORTED_VERSIONS_FROM in DatabaseSchemaVersionService,
+                    // this list should include last version and can include previous versions without upgrade
+                    String fromVersion = databaseSchemaVersionService.getDbSchemaVersion();
+                    String toVersion = databaseSchemaVersionService.getPackageSchemaVersion();
+                    cacheCleanupService.clearCache();
+                    log.info("Upgrading ThingsBoard from version {} to {} ...", fromVersion, toVersion);
+                    databaseEntitiesUpgradeService.upgradeDatabase();
+//                    dataUpdateService.updateData();
+                    installScripts.updateResourcesUsage();
```

Comment:
Move this to `updateData`. This update is only related to 3.9.0. Also, add some comment for the data update method that is should be cleaned up after each release.